account_payment_term_partner_holiday/tests/test_partner_holiday.py

```diff
@@ -140,45 +161,72 @@ def test_invoice_payment_term_partner_1(self):
         invoice_form.payment_term_id = self.payment_term_immediate
         self.assertEqual(invoice_form.date_due, fields.Date.from_string("2021-04-02"))
         invoice_form.payment_term_id = self.payment_term_10_days
-        self.assertEqual(invoice_form.date_due, fields.Date.from_string("2021-04-02"))
+        self.assertEqual(invoice_form.date_due, fields.Date.from_string("2021-04-12"))
```

Comment:
You are still modifying previous tests

Reply:
Sorry but i haven't modify tests.
Before these changes, due date generated is not correctly because not re-compute dates (only returned +1 days according to "last" holiday period), now final result in use cases is different.

Reply:
So you are telling me that there was an error in previous tests? If so, you should split the commit with the corresponding changes and explain the issue.

Reply:
I think what @victoralmau means is that this PR changes the behavior (that's why I asked [here](https://github.com/OCA/account-payment/pull/432#pullrequestreview-712391900) if this is the best approach) and, as a result, current tests need to be changed...

Reply:
I have a short-circuit about this, as I don't conceive that the dates vary this way due to that change. Is it only on tests or it's something that is happening on the behavior of the module? If it's the first, we need to create more resilient tests. If it's the second, that behavior change should be analyzed.

Reply:
It is the second, and I agree that the behavior change should be analyzed, that is why I pinged you about it above:

> Having to recompute the due date when there is a payment term set at the end is a bit weird and can cause some cases that might be hard to understand functionally. However, I am not sure how it could be done better and maybe this is the best option. Maybe @pedrobaeza can give some input here.

This basically means that, instead of selecting the first available date, the module now selects the first available one that matches the payment term "restrictions" (last day of the month, 5th day of the month, etc.)

As I said, I am not sure this suggestion from @victoralmau  is the best way of doing this, as it has a pretty big functional impact, so you will say what you prefer for it...

Reply:
Well, we are adding here an improvement for taking into account payment days, but that shouldn't impact on the previous tests suite and data, that didn't have such data to be altered, so I still don't understand why the change.

Reply:
I have modified the description with some examples to try to explain some use cases.

Yes, some use cases now have change expected due date (A and B) but IMO is the only way to get allways valid due date (C and D).